File: GaniOSApp/GanClient/GanClient/Gan/Service/MessageService.swift
//
//  MessageService.swift
//  GanClient
//
//  Created by xattacker on 2020/11/25.
//

import Foundation


public final class MessageService: ServiceFoundation
{
    public func sendMessage(_ receiver: String, message: String) -> Bool
    {
        var succeed = false
        
        let buffer = BinaryBuffer()
        buffer.writeString(receiver)
        buffer.writeString(message)
        
        if let response = self.send(FunctionType.send_sms, request: buffer.data), response.result
        {
            succeed = true
        }
        
        return succeed
    }
}

File: GanAndroidApp/GanClient/src/main/java/com/xattacker/gan/service/MessageService.kt
package com.xattacker.gan.service

import android.util.Log
import com.xattacker.binary.BinaryBuffer
import com.xattacker.gan.GanAgent
import com.xattacker.gan.data.FunctionType


class MessageService internal constructor(agent: GanAgent) : ServiceFoundation(agent)
{
    fun sendMessage(receiver: String, msg: String): Boolean
    {
        var result = false
        try
        {
            val buffer = BinaryBuffer()
            buffer.writeString(receiver)
            buffer.writeString(msg)

            val response = send(FunctionType.SEND_SMS, buffer.data)
            if (response != null)
            {
                result = response.result
            }
        }
        catch (ex: Exception)
        {
            Log.i("aaa", "aere $ex")
        }

        return result
    }
}